src/components/Table/Table.tsx

```diff
@@ -0,0 +1,351 @@
+/**
+ * Copyright 2023 Mia srl
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/* eslint-disable max-len */
+
+import { Table as AntTable, Skeleton } from 'antd'
+import { HTMLAttributes, ReactElement } from 'react'
+
+import { ColumnType, ExpandableConfig, GenericRecord, Layout, Locale, Pagination, RowSelection, Scroll, Size, UserAction } from './Table.types'
+import styles from './Table.module.css'
+
+const { Auto } = Layout
+const { Middle } = Size
+const { table } = styles
+
+export type TableProps<RecordType extends GenericRecord> = {
```

Comment:
Can it make sense to create prop-specific files given the large size for documentation?

Reply:
It would actually. I think we should align other components too. Can we address this issue in a specific refactor merge request?

Reply:
https://github.com/mia-platform/design-system/issues/235

Reply:
We can do this in an ad hoc pr